redash/tasks.py

```diff
@@ -357,18 +357,20 @@ def check_alerts_for_query(self, query_id):
                 logger.debug("Skipping notification (previous state was unknown and now it's ok).")
                 continue
 
-            # message = Message
-            html = """
-            Check <a href="{host}/alerts/{alert_id}">alert</a> / check <a href="{host}/queries/{query_id}">query</a>.
-            """.format(host=base_url(alert.query.org), alert_id=alert.id, query_id=query.id)
-
-            notify_mail(alert, html, new_state, app)
-
-            if settings.HIPCHAT_API_TOKEN:
-                notify_hipchat(alert, html, new_state)
-
-            if settings.WEBHOOK_ENDPOINT:
-                notify_webhook(alert, query, html, new_state)
+            host = base_url(alert.query.org)
+            
+            for subscription in alert.subscriptions:
+                try:
+                    if subscription.destination:
+                        subscription.destination.notify(alert.id, query.id, subscription.user.id, new_state, app, host) 
+                    else:
+                        # For backwards compatibility with prior alert system that was primarily email-based
+                        from redash.destinations import get_destination
```

Comment:
If this is just for backwards compatibility, let's handle this in the migrations:

- Create default email NotificationDestination.
- Assign it to all existing subscriptions.